addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -0,0 +1,178 @@
+	Provisioner                   = "openshift-storage.rbd.csi.ceph.com"
+)
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	klog.Infof("Running MirrorPeer reconciler on  spoke cluster")
+	// Fetch MirrorPeer for given Request
+	var mirrorPeer multiclusterv1alpha1.MirrorPeer
+	err := r.HubClient.Get(ctx, req.NamespacedName, &mirrorPeer)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.Info("Could not find MirrorPeer. Ignoring since object must have been deleted")
+			return ctrl.Result{}, nil
+		}
+		klog.Error(err, "Failed to get MirrorPeer")
+		return ctrl.Result{}, err
+	}
+	scr := r.getStorageClusterDetails(&mirrorPeer)
+
+	err = r.enableCSIOmapGenerator(ctx, scr.Namespace)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to start omap-generator for rook: %v", err)
+	}
+
+	err = r.enableMirroring(ctx, scr.Name, scr.Namespace)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to enable mirroring the storagecluster %q in namespace %q in managed cluster. Error %v", scr.Name, scr.Namespace, err)
+	}
+
+	err = r.createVolumeReplicationClass(ctx, &mirrorPeer)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to create volumereplicationclass: %v", err)
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *MirrorPeerReconciler) getStorageClusterDetails(mp *multiclusterv1alpha1.MirrorPeer) *multiclusterv1alpha1.StorageClusterRef {
+	for _, v := range mp.Spec.Items {
+		if v.ClusterName == r.SpokeClusterName {
+			return &v.StorageClusterRef
+		}
+	}
+	return nil
+}
+
+func (r *MirrorPeerReconciler) enableMirroring(ctx context.Context, storageClusterName string, namespace string) error {
+	var sc ocsv1.StorageCluster
+	err := r.SpokeClient.Get(ctx, types.NamespacedName{
+		Name:      storageClusterName,
+		Namespace: namespace,
+	}, &sc)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("could not find storagecluster %q in namespace %v: %v", storageClusterName, namespace, err)
+		}
+		return err
+	}
+	sc.Spec.Mirroring.Enabled = true
+	return r.SpokeClient.Update(ctx, &sc)
+}
+
+func (r *MirrorPeerReconciler) enableCSIOmapGenerator(ctx context.Context, namespace string) error {
+	var rcm corev1.ConfigMap
+	err := r.SpokeClient.Get(ctx, types.NamespacedName{
+		Name:      RookConfigMapName,
+		Namespace: namespace,
+	}, &rcm)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("could not find rook-ceph-config-map: %v", err)
+		}
+		return err
+	}
+
+	rcm.Data[RookCSIEnableKey] = "true"
+	rcm.Data[RookVolumeRepKey] = "true"
+	return r.SpokeClient.Update(ctx, &rcm)
```

Comment:
We need to make sure that we don't alter other fields. Can you add unit test for this?